lib/sensu/client/process.rb

```diff
@@ -496,8 +496,11 @@ def resume
       # deregistration check result if configured to do so.
       def stop
         @logger.warn("stopping")
+        last_state = @state
         pause
-        deregister if @settings[:client][:deregister] == true
+        if @settings[:client][:deregister] == true && last_state != :initializing
```

Comment:
This fixes a bug where deregistration would trigger even when the client has not yet connected to the transport.